pkg/metrics/exporters/common/common.go

```diff
@@ -0,0 +1,63 @@
+package common
+
+import (
+	"sync"
+
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/metrics/exporters/view"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+var (
+	opts            []metric.Option
+	res             *resource.Resource
+	mutex           sync.Mutex
+	requiredReaders int
+)
+
+// SetRequiredReaders sets the number of required readers for the MeterProvider.
+func SetRequiredReaders(num int) {
+	requiredReaders = num
+}
+
+// AddReader adds a reader to the options and updates the MeterProvider if the required conditions are met.
+func AddReader(opt metric.Option) {
```

Comment:
api nits: if we keep this structure, it may be useful to at least spell out the "required conditions". Alternatively splitting this into a `AddReader` and `SetMetricsProvider ` provider a cleaner/ more concise api